craft_parts/packages/rpm.py

```diff
@@ -0,0 +1,200 @@
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Support for RPM files."""
+
+import functools
+import logging
+import os
+import subprocess
+from typing import List, Set
+
+from craft_parts.utils import os_utils
+
+from . import errors
+from .base import BaseRepository
+
+logger = logging.getLogger(__name__)
+
+
+class RPMRepository(BaseRepository):
+    """Repository management for RPM packages."""
+
+    @classmethod
+    def configure(cls, application_package_name: str) -> None:
+        """Set up yum options and directories.
+
+        XXX: method left out of RPMRepository's MVP; it should be implemented
+        in the future to allow configuring the yum system.
+        """
+        logger.debug(
+            "Called not implemented method 'RPMRepository.configure' with name %r",
+            application_package_name,
+        )
+
+    @classmethod
+    def get_package_libraries(
+        cls, package_name: str
+    ) -> Set[str]:  # pylint: disable=unused-argument
+        """Return a list of libraries in package_name.
+
+        XXX: method left out of RPMRepository's MVP; cannot return a sane default so
+        raising an error to ensure it's not used.
+        """
+        raise NotImplementedError("Functionality not yet provided by RPMRepository.")
+
+    @classmethod
+    def get_packages_for_source_type(cls, source_type: str) -> Set[str]:
+        """Return a list of packages required to to work with source_type."""
+        if source_type == "bzr":
+            packages = {"bzr"}
+        elif source_type == "git":
+            packages = {"git"}
+        elif source_type == "tar":
+            packages = {"tar"}
+        elif source_type in ["hg", "mercurial"]:
+            packages = {"mercurial"}
+        elif source_type in ["svn", "subversion"]:
+            packages = {"subversion"}
+        elif source_type == "rpm2cpio":
+            # installed by default in CentOS systems
+            packages = set()
+        elif source_type == "7zip":
+            packages = {"p7zip"}
+        else:
+            packages = set()
+
+        return packages
+
+    @classmethod
+    @functools.lru_cache(maxsize=1)
+    def refresh_packages_list(cls) -> None:
+        """Refresh the list of packages available in the repository."""
+        # Return early when integration testing
+        if os.getenv("CRAFT_PARTS_PACKAGE_REFRESH", "1") == "0":
+            return
+
+        try:
+            cmd = ["yum", "update", "-y"]
```

Comment:
I guess `dnf` doesn't come installed by default in centos 7?

Reply:
it does not

Reply:
No, and when I read a little about it it was a little hacky to install, so I just moved forward with `yum`.